src/blockdata.rs

```diff
@@ -2,18 +2,20 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
+use crate::serialize::HexStrVisitor;
 use bitcoin_hashes::{sha256d, Hash};
-use serde::{Deserialize, Serialize};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
 use std::fmt::Debug;
 
 pub mod hash {
     use crate::errors::Error;
-    use serde::{Deserialize, Serialize};
+    use crate::serialize::HexStrVisitor;
+    use serde::{Deserialize, Deserializer, Serialize, Serializer};
     use std::fmt::Debug;
 
     /// This is hash value container struct.
     /// This struct assumes porting value from sha256d::Hash.
-    #[derive(Serialize, Deserialize, PartialEq)]
+    #[derive(PartialEq, Clone, Copy)]
     pub struct Hash([u8; 32]);
```

Comment:
This structure is strange. Why define your own? Furthermore, the name Hash is abstract. You can guess SHA256 from the size definition, but you should clearly name what you support.

Reply:
Probably, this is for implementing `serde::Serialize` and `serde::Deserialize`.

Reply:
You mean `Sha256dHash` is better?

Reply:
I've changed the name to `SHA256Hash`